### DISCUSSION TITLE
Documentation Search Not Indexing Code Blocks

### DISCUSSION THREAD

**Discussion by freshleafmedia** [area17/twill#1367] [General]
Hello!

I've using Twill for a while now (it's awesome by the way) but am finding it hard to get the information I need. Most of the time the information required is in the documentation but the search fails to find many things I would expect it to and I think this is down to it not indexing code blocks.

For example searches for the following terms don't give the results I'd hope to expect:

- `indexOptions` or `indexColumns` - hoped to find https://twill.io/docs/crud-modules/controllers.html
- `@formField('input')` - hoped to find https://twill.io/docs/form-fields/input.html
- `@formConnectedFields` - hoped to find https://twill.io/docs/form-fields/conditional-fields.html
- `block_preview_render_childs` - hoped to find https://twill.io/docs/block-editor/default-configuration.html

This wasn't such a problem when it was a single page because I could use the browsers search. I'm not suggesting we go back to a single page layout because an Algolia powered search should beat a browser search hands down.

Is this something that can be changed?

**Comment by haringsrob** [comment]
I agree, I also often have this. What I then fall back to is google search:

`site:https://twill.io/docs/ indexOptions` for example. But it would be 10x better if this worked via the search bar.

**Reply by freshleafmedia** [reply]
One work around I've taken to using is to search `vendor/area17/twill/docs/src/` using the IDE

**Comment by ifox** [comment]
Definitely agree. Looked into it quickly: https://docsearch.algolia.com/docs/legacy/faq/#can-you-index-code-samples

**Comment by ifox** [comment]
Here's Twill's config: https://github.com/algolia/docsearch-configs/blob/master/configs/twill.json

**Comment by haringsrob** [comment]
So they do recommend not doing it. We can give it a try by including the pre tags? Can always revert it later if it indeed add's noise.
